crates/cast/src/cmd/run.rs

```diff
@@ -287,6 +295,38 @@ impl RunArgs {
     }
 }
 
+pub async fn fetch_contracts_bytecode_from_trace(
+    provider: &RootProvider<AnyNetwork>,
+    result: &TraceResult,
+) -> Result<HashMap<Address, Bytes, RandomState>> {
+    let mut contracts_bytecode = HashMap::new();
+    if let Some(ref traces) = result.traces {
+        let addresses = gather_trace_addresses(traces);
+        for address in addresses {
+            let code = provider.get_code_at(address).await?;
+            if !code.is_empty() {
+                contracts_bytecode.insert(address, code);
+            }
+        }
```

Comment:
this fetches codes code sequentially, we could fetch   concurrently

also, should this fail or should we simply skip errors here?

Reply:
Done for `join_all`, as well as warn on error (I think I did it correctly?)